### ISSUE TITLE
The Climb Extremely Choppy

### ISSUE THREAD

**Issue by ziggyzerda** [LibreVR/Revive#403]
So I'm trying to play The Climb as I love these type of VR games. However every time I play it, no matter the graphic settings in game, it's very laggy and choppy and takes no time to give me headaches and nauseous. 

My computer specs are way above capable to run this game.

Intel i5-3330 @ 3GHz
16GB RAM
GeForce GTX 1060 6GB

How can I stop it from causing this issue?

**Comment by LibreVR** [comment]
Try enabling "Always on Reprojection" in the SteamVR developer settings. I don't recommend you keep this one, but it will tell us whether the CPU is the bottleneck or not. Let me know if it improves the situation.

**Comment by ziggyzerda** [comment]
Still showing the same issue. When I move my head from side the side the menu and background will hang/lag/jump. Does so with Reprojection enabled and disabled. It's showing perfect framerate in general, through the monitor, so I'm assuming the the lag is only displayed through the headset.

**Comment by dedwnl** [comment]
How are you turning Älways On" reprojection on and off? I am using Advanced OpenVR Settings, and if I turn reprojection and ASW on/off individually or together, nothing happens. However, the "Always On" option is a different box and requires a SteamVR restart. This completely fixed performance for me and I went from 1.2 SS with Medium to 1.5 on High. It's super smooth.

N.B. It did not fix the PC screen flickering when the ingame player sweats (see my other issue)

**Comment by ziggyzerda** [comment]
I have installed and ran the "Advanced OpenVR Settings" as I did not know this was a thing. I found what you were talking about and it was already turned on (the "Always On"). It does need a restart in the advanced options, but didn't prompt me to in the regular GUI on my computer. Still having the same issue. It runs smooth as silk off of my monitor, but in my headset it is extremely choppy to where I can't play it without getting a headache. I am also noticing the issue of the "flickering" you're getting, though it's rare.

**Comment by LibreVR** [comment]
Ok, thanks for confirming it is not a CPU issue, please turn "Always On Reprojection" back off again and follow these instructions to report performance data: https://github.com/LibreVR/Revive/issues/353#issuecomment-265434808

Please also include your `steamvr.vrsettings` from `C:\Program Files (x86)\Steam\config\steamvr.vrsettings`, rename it to `steamvr.txt` so you can upload it here.

**Comment by jamax** [comment]
Im' having a similar issue. 
I can't go above ~1.3SS before experiencing some stuttering, specially at 1.4SS.
I'm  running an i7-6700K with 1080gtx, and from what I read, I should be able to reach about 1.8SS.

Would you like a report from me too, or is it just normal that it stutter after 1.3SS ?

**Comment by jamax** [comment]
I may have a similar issue, as the game stutter too fast compared to what I should be able to reach.
Running it with a 1080gtx + i7-6700k, and I can't go above 1.3SS setting witthout experiencing stuttering (I tried with every retroprojection, in-game settings etc...).
For example on this Oculus post, people says I should be able to reach somewhere between 1.8 and 2.0 SS without experiencing stuttering:
https://www.reddit.com/r/oculus/comments/5hza07/shout_out_to_oculus_and_the_climb_team_for/

Would you like any report or can't I expect more ?

**Comment by LibreVR** [comment]
@dedwnl I can't imagine that the Climb, which is basically "Crysis for VR" will preform very well with super sampling. Most VR games on the Vive use Unity, which is generally a lot more light-weight and allow you to do more super-sampling.

So please, when using Revive with AAA titles, set your SS to 1.0.

**Comment by ziggyzerda** [comment]
Here is the information you are requesting. Thank you.

![untitled](https://cloud.githubusercontent.com/assets/24688120/21444005/3b507220-c877-11e6-8f01-1c24dc5a3f1d.png)

[steamvr.txt](https://github.com/LibreVR/Revive/files/670067/steamvr.txt)

[VRFrames.txt](https://github.com/LibreVR/Revive/files/670068/VRFrames.txt)

**Comment by LibreVR** [comment]
It looks like your GPU is completely overloaded try turning down the graphics settings in The Climb.

**Comment by ziggyzerda** [comment]
The graphics setting has been tested in various manners. High, medium, and lowest as it can go. Still showing same issue, when my graphic card I have surpasses "Recommended Requirements".

**Comment by jamax** [comment]
I can understand how you feel @ziggyzerda, howeverthis is mostly due to how poorly TheClimb (via CryEngine) has been optimized.

For instance, I'm running TheClimb through Revive too, with an  i7-6700k and a 1080gtx.
My GPU and CPU are at least 30% superior to your config. I can however not go above medium settings with 1.3SS.

If everything else has fails, you can still try to:
- lower down your compositor settings to ~0.5 just to see (it seems it interacts with the in-game menu)
- disable all retroprojection , seems to work better for me
- remove FPS warning, which seems to cause trouble for some people
- disable all anti-aliasing

If nothing of this works, I will be sorry for you, but I ain't sure it will be Revive fault. There are a lot of post on reddit about how poorly the climb is optimized, even when played directly on Oculus.

For instance : https://www.reddit.com/r/oculus/comments/5hsqvm/keep_getting_low_frame_rate_warnings_in_the_climb/ , and there are many other posts/comment about issues on this game.

I personnaly bought and try "TheClim", "I expect you to die" and get the free "Bullet Train", and the climb is by far the worst looking and most demanding game, got no problem with the others.

TL;DR: blame the game, not revive

**Comment by LibreVR** [comment]
@ziggyzerda If you can send the same performance data with the lowest graphics settings active, then I may be able to get a better idea of where the bottleneck is. I also see a lot of CPU spikes, which is something Revive *can* do something about, but that data is drowned out in the continuous spike on the GPU.

**Comment by unclespode** [comment]
I'm having the same issue as described here. I am on the default graphics settings in game. I have an i5 / GTX 960 OC.

By lowering the graphics setting all the way to the bottom - everything was fine. I'll fiddle until I find something that looks good. 

The way I would describe it - if I didn't move my head and just looked in the same place - the frame rate was good, but as I turned my head it would kind of jump back and forth.

**Comment by sthaber** [comment]
This seems to be a dupe of  #427. I posted my performance data there.

**Comment by Armada651** [comment]
Other way around, #427 is a dupe of this one.

**Comment by Armada651** [comment]
It seems likely that The Climb is running into a bottleneck due to the "late start" mechanism. We are supposed to wait for late start when we request new tracking data, but due to multi-threading limitations in OpenVR there are cases where I have to wait a lot earlier.

I'll try and get around those limitations, but I hope OpenVR will improve the multi-threading situation.

**Comment by Armada651** [comment]
Scratch that, when I'm out of the Tutorial it's looking more like a GPU bottleneck, instead of the CPU bottleneck I outlined in my previous comment.

**Comment by Armada651** [comment]
Seems like disabling both asynchronous and interleaved re-projection fixes the issue on my PC. Can anyone else test The Climb with both reprojection options turned off?

**Comment by sthaber** [comment]
It still performs poorly for me with all reprojection turned off. I posted debugging info in the other issue for this, #427.

**Comment by Armada651** [comment]
@sthaber Try turning on the option `Enable Always-on Reprojection` in the Developer tab of the SteamVR settings. If that makes the game perform better then that proves it's a CPU bottleneck.

**Comment by sthaber** [comment]
With both reprojection modes (interleaved / async) enabled and always-on reprojection enabled, the headset seemed locked at 90 FPS, although the motion delay was much much higher, making the game a bit nauseating to play. It certainly fixes the judder, however.

I also tried with both reprojection modes (interleaved / async) _disabled_ and always-on reprojection turned on, but the Vive display stayed dark and wouldn't come on. I guess I don't know what I expected. How can you force reprojection without a reprojection method enabled?

Why is it a CPU bottleneck when forced reprojection is on? I was under the impression that reprojection is a fairly cheap transform that can be applied to an already-rendered frame. It's used to take load off the GPU when it can't keep up with the target render times. Wouldn't that mean that it's GPU bottleneck, if forcing reprojection is required to hit the target framerate?

**Comment by Armada651** [comment]
> Wouldn't that mean that it's GPU bottleneck, if forcing reprojection is required to hit the target framerate?

Nope, always-on reprojection doesn't generate new frames, so the load on the GPU is the same. Normally OpenVR forces the CPU into a "late start", which means it will block the CPU until a few milliseconds before the GPU wants to render a new frame. This is intended to ensure that the CPU sends the latest pose data to the GPU.

By enabling always-on reprojection OpenVR will disable "late start" and instead compensate for the fact that the pose data is a lot older always applying that latest head pose using reprojection. Thus freeing up a bunch of time for the CPU to generate each frame. However due to the fact reprojection can only correct for orientation this is a bit more nauseating as you said.

**Comment by sthaber** [comment]
Got it! Thanks for the explanation.

**Comment by unclespode** [comment]
Contrary to what I suggested earlier - lowering the graphics settings does alleviate the issue - but does not solve it. Even with everything on low - this tracking issue is quite jarring.

**Comment by Armada651** [comment]
I've done a lot of research on The Climb and I've found the reason why reprojection is bugged in this game.

Revive assumes that applications will request tracking data for the next frame, render the frame based on that tracking data and then submit the frame.

Instead, The Climb requests tracking data for the frame after the next frame, and then submits a frame based on the tracking data returned from a previous request.

This means the CPU is running one frame ahead of the GPU, which increases latency but allows for better GPU utilization (doesn't have to wait for the CPU as much).

Unfortunately this is not something that's allowed in OpenVR currently. And reprojection will fail when it receives a frame that wasn't rendered with the latest tracking data.

**Comment by Nordomus** [comment]
Nothing worked for me :( I'm on GTX 1070 and i5 4690k and I can't get it to run smoothly :(

**Comment by Armada651** [comment]
@Nordomus I'm currently working on this issue, hopefully I'll have some progress to report soon.

**Comment by Armada651** [comment]
@Nordomus Please try the latest release: https://github.com/LibreVR/Revive/releases/tag/1.1

It should improve performance a little bit, but you may still need to enable Always-on Reprojection in the SteamVR developer settings. (Turn if off again when you're done playing The Climb)

You probably also need to mess with enabling/disabling Asynchronous and/or Interleaved Reprojection.

**Comment by rudiemcska** [comment]
Updating to 1.1 helped a bit. I found it playable with "Allow asynchronous re projection" enabled, "Allow interleaving reprojection" disabled and of course "enable always on reprojection" enabled. With those settings the main menu stuttered more then the first non-training climbing level so try and look past that.

GTX 1080 here, I found changing the in game settings didn't seem to effect how severely it would stutter in game or menu.

**Comment by Stuporsteve** [comment]
Hey, I have had The Climb for over a month now (since I bought the Oculus). It has run perfectly, and I often like to do a climb first thing in the morning. I had done one Friday, April 21st, and it was fine, then Saturday morning it was choppy. Like any other issue that crops up like this, I immediately shut everything down and restarted. Nothing helped, and all of the other games ran perfectly. I am inclined to believe that a new update for the Climb has changed something.

I haven't yet tried any of the above-mentioned tips, but I don't believe my hardware would have any issues. I bought a high-end computer in anticipation of the Oculus specifically, and especially when everything was running fine just a couple of days ago.

CPU - Intel Core i7 6800K
GPU - GeForce GTX 1070

**Comment by paulopsx2** [comment]
anyone have a SAVE GAME file with 100% finish game? or something else... my files are corrupt ( cyclicing error on HD ), tks

**Comment by Stuporsteve** [comment]
I booted up my Oculus this morning for the first time after a few days, there was a software update that needed to install. The Climb was the first thing I started up once the update was complete and everything ran smoothly. I don't know what this means for the Vive or any other VR platform, but the Oculus is running perfectly find once more.

**Comment by mikegomak** [comment]
@Stuporsteve I'm confused, why are you using oculus via revive?

**Comment by Nordomus** [comment]
@mikegomak I'm confused, how else would you use it?

**Comment by mikegomak** [comment]
I thought this thread is for issues relating to revive - the (amazing) software that allows legally purchased Oculus games to run on the Vive. If you are running an Oculus Rift then Im not sure this is the correct thread for your issue.

**Comment by Armada651** [comment]
@mikegomak @Nordomus He didn't say he was using his Oculus Rift with Revive.

**Comment by Armada651** [comment]
Fixed in the latest release: https://github.com/LibreVR/Revive/releases/tag/1.1.12

**Comment by sthaber** [comment]
Great work, thank you!! Look forward to trying this soon.

**Comment by Armada651** [comment]
Remember to use the SteamVR beta and enable Asynchronous Reprojection.

**Comment by zealousappex** [comment]
Hi,

I just want to help anyone else having a issue with The Climb and Revive this is what I did and I am able to in game setting re-sample to 1.5 now without glitching or issues.

1. CLOSE any program that you are using to OVERCLOCK, in my case I closed Afterburner. 

2. Turn ON"Enable Always-on Reprojection".

3. Turn OFF all of these: "Enable Advance Supersampling Filter", "Allow Asynchronous Reprojection" and "Allow Interleaved Reprojection"
